prov/gni/test/cntr.c

```diff
@@ -797,3 +797,89 @@ Test(cntr_mt, write_wait)
 
 	dbg_printf("done\n");
 }
+
+void *do_add_cntr(void *arg)
+{
+	int i = 0, ret, iters = ((int *)arg)[0];
+
+	for (; i < iters; i++) {
+		ret = fi_cntr_add(ep_write_cntrs[0], 1);
+		cr_assert(ret == FI_SUCCESS);
+	}
+
+	return NULL;
+}
+
+void *do_add_err_cntr(void *arg)
+{
+	int i = 0, ret, iters = ((int *)arg)[0];
+
+	for (; i < iters; i++) {
+		ret = fi_cntr_adderr(ep_write_cntrs[0], 1);
+		cr_assert(ret == FI_SUCCESS);
+	}
+
+	return NULL;
+}
+
+Test(cntr_mt, set_add_read_cntr)
+{
+	int iters = 128, nthreads = 4, i, ret;
```

Comment:
I agree. Just add a single threaded test for each of the set/add and it's looking good.